src/ImageCaptureCore/ICCameraDevice.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using Foundation;
+using ObjCRuntime;
+
+namespace ImageCaptureCore {
+	partial class ICCameraDevice {
+
+		public delegate void DidReadDataDelegate (NSData data, ICCameraFile file, NSError error);
+
+		public void RequestReadDataFromFile (ICCameraFile file, long offset, long length, DidReadDataDelegate callback)
+		{
+			var actionObject = new DidReadDataFromFileAction (callback);
+			RequestReadDataFromFile (file, offset, length, actionObject, Selector.FromHandle (Selector.GetHandle ("didReadData:fromFile:error:contextInfo:")), IntPtr.Zero);
```

Comment:
It's better to use the constant since we have it:

```suggestion
			RequestReadDataFromFile (file, offset, length, actionObject, Selector.FromHandle (Selector.GetHandle (DidReadDataFromFileAction.CallbackSelector)), IntPtr.Zero);
```

and the same for the other cases below.